ex5/ex5.c

```diff
@@ -16,7 +16,29 @@ char* msg3 = "hello world #3";
 
 int main(void)
 {
-    // Your code here
-    
+    char inbuf[MSGSIZE];
+    int p[2]; 
+    if (pipe(p) < 0) {
+        fprintf(stderr, "pipe failed\n");
+        exit(1);
+    }
+
+    int pid = fork();
+    if (pid < 0) {    // fork failed; exit
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+    } else if (pid == 0) {
+        write(p[1], msg1, MSGSIZE);
+        write(p[1], msg2, MSGSIZE);
+        write(p[1], msg3, MSGSIZE);
+        close(p[1]);
+    } else {
+        for (int i = 0; i < 3; i++) {
+            read(p[0], inbuf, MSGSIZE);
```

Comment:
perfect. for readability's sake, you can `close(p[1])` on the process where you're reading, and `close(p[0])` between lines 30-31. it's fine without doing it though as long as you don't call on them (like you're doing here)